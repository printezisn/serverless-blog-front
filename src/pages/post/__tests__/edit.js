import React from "react"
import renderer from "react-test-renderer"

import EditPage from "../edit"

jest.mock("@reach/router", () => {
    const location = {
        search: "?id=test-id",
    }

    return {
        Location: props => <p>{props.children({ location })}</p>,
    }
})
jest.mock("../../../components/layout", () => "Layout")
jest.mock("../../../components/post", () => {
    return {
        PostLoader: "PostLoader",
        EditPost: "EditPost",
    }
})

describe("Edit page", () => {
    it("renders correctly", () => {
        const editPage = renderer.create(<EditPage />)

        expect(editPage.toJSON()).toMatchSnapshot()
    })

    it("renders the edit post form correctly", () => {
        const post = { title: "Test Title" }
        const editPage = renderer.create(<EditPage />)
        const form = editPage.root.findByType("PostLoader").props.onRender(post)

        expect(form).toMatchSnapshot()
    })
})
