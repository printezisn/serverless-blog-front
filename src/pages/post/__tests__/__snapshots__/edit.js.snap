// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit page renders correctly 1`] = `
<Router>
  <Layout>
    <header>
      <h2
        className="title is-3 has-text-centered"
      >
        Edit Post
      </h2>
    </header>
    <PostLoader
      onRender={[Function]}
    />
  </Layout>
</Router>
`;

exports[`Edit page renders the edit post form correctly 1`] = `
<EditPost
  post={
    Object {
      "title": "Test Title",
    }
  }
/>
`;
